src/app/components/modal-window/modal-window.component.ts

```diff
@@ -1,22 +1,34 @@
-import { Component, HostListener, Input, OnInit } from '@angular/core';
-
-import { AuthService } from 'src/app/services/auth.service';
+import { Component, HostListener, Input, OnDestroy, OnInit } from '@angular/core';
 import { ModalWindowService } from './modal-window.service';
-import { User } from 'src/app/models/user.model';
 
+//all console logs will be delete, when functions will be done
 @Component({
   selector: 'app-modal-window',
   templateUrl: './modal-window.component.html',
   styleUrls: ['./modal-window.component.scss'],
 })
-export class ModalWindowComponent implements OnInit {
-  constructor(private modalWindowService: ModalWindowService, private authService: AuthService) {}
+export class ModalWindowComponent implements OnInit, OnDestroy {
+  constructor(private modalWindowService: ModalWindowService) {}
 
-  public vision: boolean;
+  private vision: any;
+  private form: any;
+  public whatIsTheForm: any;
```

Comment:
Why do we need this field? Modal shouldn't care about content within

Reply:
vision and form are for destroy subscribers
whatIsTheForm is boolean. this is for send or get true/false login form